pkgs/applications/system/howdy/default.nix

```diff
@@ -0,0 +1,89 @@
+{ stdenv
+, lib
+, bzip2
+, fetchFromGitHub
+, fetchurl
+, fmt
+, gettext
+, inih
+, installShellFiles
+, libevdev
+, meson
+, ninja
+, pam
+, pkg-config
+, python3
+} @ args:
+let
+  data = import ./sources.nix args;
+in
+stdenv.mkDerivation {
+  pname = "howdy";
+  version = "unstable-2023-02-28";
+  inherit (data) src;
+
+  # fix paths
+  patches = [ ./howdy.patch ];
+
+  postPatch =
+    let
+      howdypath = "${placeholder "out"}/lib/security/howdy";
+    in
+    ''
+      substituteInPlace howdy/src/cli/add.py --replace "@PATH@" "${howdypath}"
+      substituteInPlace howdy/src/cli/config.py --replace '/bin/nano' 'nano'
+      substituteInPlace howdy/src/cli/test.py --replace "@PATH@" "${howdypath}"
+
+      substituteInPlace howdy/src/pam/main.cc \
+        --replace "python3" "${data.py}/bin/python" \
+        --replace "/lib/security/howdy/compare.py" "${howdypath}/compare.py"
+
+      substituteInPlace howdy/src/compare.py \
+        --replace "/lib/security/howdy" "${howdypath}" \
+        --replace "@PATH@" "${howdypath}"
+    '';
+
+  nativeBuildInputs = [ bzip2 installShellFiles meson ninja pkg-config ];
+  buildInputs = [ data.py fmt gettext inih libevdev pam ];
+
+  # build howdy_pam
+  preConfigure = ''
+    cd howdy/src/pam
+    export DESTDIR=$out
```

Comment:
Using `DESTDIR` is yucky, see https://github.com/NixOS/nixpkgs/issues/65718.
Here it appears to be used as a workaround for a [hardcoded `/lib` as an installation path](https://github.com/boltgolt/howdy/blob/30728a6d3634479c24ffd4e094c34a30bbb43058/howdy/src/pam/meson.build#L22). That should really be fixed upstream but in the meanwhile, could you add a comment with explanation?